src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -58,9 +58,33 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
   const networkPolicyNames = new Set<string>();
 
   for (const policy of networkPolicy) {
-    // remoteGenerated cannot be combined with remoteNamespace or remoteSelector
-    if (policy.remoteGenerated && (policy.remoteNamespace || policy.remoteSelector)) {
-      return req.Deny("remoteGenerated cannot be combined with remoteNamespace or remoteSelector");
+    // Ensure only valid combinations of remoteGenerated, remoteNamespace, remoteSelector, and remoteCidr
+    if (policy.remoteGenerated) {
+      if (policy.remoteNamespace || policy.remoteSelector || policy.remoteCidr) {
+        return req.Deny(
+          "remoteGenerated cannot be combined with remoteNamespace, remoteSelector, or remoteCidr",
+        );
+      }
+    } else if (policy.remoteNamespace && policy.remoteSelector) {
+      if (policy.remoteGenerated || policy.remoteCidr) {
+        return req.Deny(
+          "remoteNamespace and remoteSelector cannot be combined with remoteGenerated or remoteCidr",
+        );
+      }
+    } else if (policy.remoteNamespace) {
+      if (policy.remoteGenerated || policy.remoteCidr) {
+        return req.Deny("remoteNamespace cannot be combined with remoteGenerated or remoteCidr");
+      }
+    } else if (policy.remoteSelector) {
+      if (policy.remoteGenerated || policy.remoteCidr) {
+        return req.Deny("remoteSelector cannot be combined with remoteGenerated or remoteCidr");
+      }
```

Comment:
nit: I'd be in favor of collapsing these 3 for code simplicity...
```suggestion
    } else if (policy.remoteNamespace || policy.remoteSelector) {
      if (policy.remoteGenerated || policy.remoteCidr) {
        return req.Deny(
          "remoteNamespace and remoteSelector cannot be combined with remoteGenerated or remoteCidr",
        );
      }
```

If we wanted to make the deny message super explicit about which one was used that clashed I think we could probably conditionally modify the deny message.

Reply:
accepting your proposed changes here, we can come back later and add more explicit logging if it's an issue. IMO